### ISSUE TITLE
Milestone 3

### ISSUE THREAD

**Issue by michal-szostak** [cyfronet-fid/sat4envi#253]
This issue has been created in order to list all the things which are required for milestone 3.
When relevant issue is created it should be linked in the description here.

## Front End

* [x] Connect map state (open layers) to the store #251  
* [x] Sync view parameters between store and URL #252
* [x] Create Module for Modals #262
* [x] Refactor code to reflect new naming scheme (Products, Overlays, Scenes) #258
* [ ] Loading state for the sidebar #250-> M4
* [ ] Connect to TZ aware backend #347-> M4
* [ ] Search styling-> M4
* [ ] Registration UI / styling (account activation step)-> M4
* [ ] Invite UI / styling (account activation step -> might be the same as previous step with some parametrisation)-> M4
* [ ] Timeline loading state, general styling-> M4
* [x] Generating PNG image from canvas #257
* [x] Generating PDF report from canvas #266
* [x] Raport generation modal #266
* [x] Share view modal #274
* [x] Save view modal, Saved configurations modal #273
* [x] Drop Down (show / hide) for ZK options #254
* [x] Form module (create reusable form components, with error & loading handling) [ Blocked by: #255 ]
* [ ] General application loading screen (now, when user navigates to app, before JS is loaded there is only a white screen

## Back End
* [x] Return download url for a scene #36
* [x] userroles + authorization #169
* [x] Lazy loading #220 
* [x] Share Link to view #249
* [x] Add caption, description, thumbnail to link sharing endpoint #321
* [x] Refactor code to reflect new naming scheme (Products, Overlays, Scenes) #231
* [x] Timezone awareness - API should consume TZ and return appropriate response #307 
* [x] HTML mail templates #311
* [x] Mail styling (current mails: Account creation, Invitation, @kmarszalek something else?) #314
* [x] Better parametrisation of the backend env variables / modes #256
* [x] Migrate API docs to springdoc-openapi #267
* [x] Endpoint & DB entity for saved views (@abacz do we want to also store map miniature there? I saw it on the previews...) #272
* [x] Disable OSIV #283
* [ ] Spring should return `401` error when user is not authenticated (now it always throws 403)-> M4
* [ ] Invite workflow-> M4
* [ ] institution contains logo, address,etc (note - logos should be optional)-> M4
* [ ] Add validation to SavedView thumbnail #286 -> M4
* [ ] Request throttling #295 -> M4

## CEPH

* [ ] Upload of training Sentinel data to CEPH #279 -> M4

## Infrastructure
* [x] (Nice to have, not required) E2E infrastructure preparation #260 (done to some extent #333)
* [ ] Email account + production instance configuration #292 -> M4

## For further consideration
* [ ] Map view is full screen, which poses problems when saving it to file (saved image contains more than what the user effectively sees, it also makes zooming in via shift-select unintuitive (part of the zoomed in area can be hidden under sidebar)
